homeassistant/components/climate/broadlink.py

```diff
@@ -0,0 +1,270 @@
+                         '"temp":20}, '
+                         '{"start_hour":23, '
+                         '"start_minute":0, '
+                         '"temp":17}]'): cv.string
+})
+
+SET_SCHEDULE_SCHEMA = vol.Schema({
+    vol.Required(CONF_WEEKDAY,
+                 default='[{'
+                         '"start_hour":6, '
+                         '"start_minute":30, '
+                         '"temp":20}, '
+                         '{"start_hour":9, '
+                         '"start_minute":0, '
+                         '"temp":17}, '
+                         '{"start_hour":12, '
+                         '"start_minute":0, '
+                         '"temp":20}, '
+                         '{"start_hour":14, '
+                         '"start_minute":0, '
+                         '"temp":17}, '
+                         '{"start_hour":18, '
+                         '"start_minute":0, '
+                         '"temp":20}, '
+                         '{"start_hour":22, '
+                         '"start_minute":30, '
+                         '"temp":17}]'): cv.string,
+    vol.Required(CONF_WEEKEND,
+                 default='[{"start_hour":8, '
+                         '"start_minute":30, '
+                         '"temp":20}, '
+                         '{"start_hour":23, '
+                         '"start_minute":0, '
+                         '"temp":17}]'): cv.string
+})
+
+SET_ADVANCED_CONF_SCHEMA = vol.Schema({
+    vol.Required(CONF_ADVANCED_CONFIG,
+                 default='{"loop_mode": 0, '
+                         '"sen": 0, '
+                         '"osv": 42, '
+                         '"dif": 2, '
+                         '"svh": 35, '
+                         '"svl": 5, '
+                         '"adj": 0, '
+                         '"fre": 1, '
+                         '"pon": 0}'): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the broadlink thermostat platform."""
+    import BroadlinkWifiThermostat
+    wifi_thermostat = BroadlinkWifiThermostat.\
+        Thermostat(config[CONF_MAC],
+                   config[CONF_HOST],
+                   config[CONF_NAME],
+                   config[CONF_ADVANCED_CONFIG],
+                   config[CONF_SCHEDULE_WEEKDAY],
+                   config[CONF_SCHEDULE_WEEKDAY],
+                   config[CONF_MIN_TEMP],
+                   config[CONF_MAX_TEMP],
+                   STATE_OFF,
+                   STATE_HEAT,
+                   STATE_AUTO
+                   )
+
+    add_devices([BroadlinkThermostat(wifi_thermostat)], True)
+
+    @asyncio.coroutine
+    def handle_set_schedule(service):
+        """Handle data for the set_schedule service call."""
+        schedule_wd = service.data.get(CONF_WEEKDAY)
+        schedule_we = service.data.get(CONF_WEEKEND)
+        wifi_thermostat.set_schedule(
+            {CONF_WEEKDAY:
+             json.loads(schedule_wd.replace("'", '"')),
```

Comment:
If we keep this service we should not allow json string for attribute value but it should be a python container that we validate totally with a voluptuous schema. Ie each key and value should be validated by the schema.